compiler-core/src/type_.rs

```diff
@@ -1579,18 +1579,39 @@ fn make_type_vars(
         .try_collect()
 }
 
-fn custom_type_accessors<A>(
+#[allow(unused_qualifications)]
```

Comment:
We should never disable the linter, best fix this.

Reply:
Without that I get this:
```
warning: unnecessary qualification                                                                                                                                                                                                     
    --> compiler-core\src\type_.rs:1582:37
     |
1582 | fn custom_type_accessors<A: Clone + std::cmp::PartialEq>(
     |                                     ^^^^^^^^^^^^^^^^^^^
     |
note: the lint level is defined here
    --> compiler-core\src\lib.rs:33:5
     |
33   |     unused_qualifications,
     |     ^^^^^^^^^^^^^^^^^^^^^
   ```
  I then remove it to get this on the next build:
  ```
PS B:\Development\gleam-lang> cargo build
   Compiling gleam-core v0.21.0-rc1 (B:\Development\gleam-lang\compiler-core)
error[E0369]: binary operation `==` cannot be applied to type `&ast::RecordConstructorArg<A>`                                                                                                                                          
    --> compiler-core\src\type_.rs:1617:30
     |
1617 |                         item == &clear_data_type
     |                         ---- ^^ ---------------- &ast::RecordConstructorArg<A>
     |                         |
     |                         &ast::RecordConstructorArg<A>
     |
help: consider further restricting this bound
     |
1582 | fn custom_type_accessors<A: Clone + std::cmp::PartialEq>(
     |                                   +++++++++++++++++++++

For more information about this error, try `rustc --explain E0369`.                                                                                                                                                                    
error: could not compile `gleam-core` due to previous error
```

Reply:
It's asking you to remove `std::cmp::`, that's the qualification. If you `cmd+.` on it in vscode it'll be able to fix it for you

Reply:
Oh sorry, will try now. I'm currently using CLion and it isn't helping with that 😅